src/cmd/ksh93/sh/path.c

```diff
@@ -1386,18 +1386,18 @@ static bool path_chkpaths(Shell_t *shp, Pathcomp_t *first, Pathcomp_t *old, Path
             }
             *cp = 0;
             m = ep ? (ep - sp) : 0;
-            if (m == 0 || (m == 6 && memcmp((void *)sp, (void *)"FPATH=", m) == 0)) {
+            if (m == 0 || (m == 6 && strncmp((void *)sp, (void *)"FPATH=", m) == 0)) {
```

Comment:
Since you're touching this please remove the pointless cast to `void *` here and below.